### ISSUE TITLE
Colon rules acting differently for same situation. (+ custom question)

### ISSUE THREAD

**Issue by freak4pc** [realm/SwiftLint#730]
Hey There :)

In many places in my code I prefer having my colons attached to the identifier, but not with a single space to the type. I think this requires a custom rule (Something like `identifier(no spaces):(multiple spaces)type`), but I'm still not fluent enough in Swiftlint to write this rule

The odd part is the current rule seems picky on this and sometimes lets these kind of statements pass... I'm not sure what's the difference between the first line that wasn't detected as a "warning" to the other two, which are. 

![screen shot 2016-07-25 at 11 38 05 am](https://cloud.githubusercontent.com/assets/605076/17095369/c79234a8-525c-11e6-81e6-c39153e259b3.png)

Also, if you could guide me on how to write a rule that will allow the following detection (meaning disabled `colon_rules` and writing something custom that allows multiple spaces *after* the colon) I'd be happy to read up on it.

Thank you!

**Comment by jpsim** [comment]
It's odd that only 2 of the 3 warnings are rendered for you.... 🤔

I can't reproduce that:

```
$ swiftlint version
0.11.0
$ cat main.swift
public class Location {
  /// Store Location Address.
  dynamic public var address:    String!

  /// Store Location City.
  dynamic public var city:       String!

  /// Store Location Country.
  dynamic public var country:    String!
}
$ swiftlint
Linting Swift files in current working directory
Linting 'main.swift' (1/1)
main.swift:3:22: warning: Colon Violation: Colons should be next to the identifier when specifying a type. (colon)
main.swift:6:22: warning: Colon Violation: Colons should be next to the identifier when specifying a type. (colon)
main.swift:9:22: warning: Colon Violation: Colons should be next to the identifier when specifying a type. (colon)
Done linting! Found 3 violations, 0 serious in 1 file.
```

---

As for making changes to rules to support this kind of style, I'm all for it!

The way I'd recommend this be done is to make `ColonRule` configurable, maybe with a `flexible_right_spacing` boolean flag that allows one _or more_ spaces to the right of the colon.

**Comment by freak4pc** [comment]
I'm down for trying to write that but I'm still not very acquainted with
how the rule mechanism works.

Gonna read a bit into it and see how well my brain works today 🤓

**Comment by freak4pc** [comment]
Well that was easier than I thought: https://github.com/realm/SwiftLint/pull/733.

Tested on my pretty massive code base and resolved the issue as expected. Let me know if you're cool with merging this solution or there are any modifications you'd like me to make. 

My config now includes

```YAML
colon:
    flexible_right_spacing: true
```

Cheers :)
Shai. 

P.S. I wanted to add another unit test for testing this specific case but wasn't completely sure on how to make that work, since `nonTriggeringExamples` and `triggeingExamples` should be different based on the configuration. Any info on that would be appreciated :)

**Comment by freak4pc** [comment]
Not sure why, the Travis build for SPM fails (it can't clone Commandant for some odd reason). 
The tests are running fine locally.
